google-cloud-spanner/src/main/java/com/google/cloud/spanner/Value.java

```diff
@@ -1795,6 +1839,29 @@ void valueToString(StringBuilder b) {
     }
   }
 
+  private static class IntervalImpl extends AbstractObjectValue<Interval> {
+
+    private IntervalImpl(boolean isNull, Interval value) {
+      super(isNull, Type.interval(), value);
+    }
+
+    @Override
+    public Interval getInterval() {
+      checkNotNull();
+      return value;
+    }
+
```

Comment:
What is the default implementation for `getAsString()`? Does that return what we would want for `Interval`? Otherwise, we should add an implementation for that as well here.

Reply:
added default implementation for `getAsString`, it also returns ISO8601 formatted Interval